org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/SonarLintPartListener.java

```diff
@@ -64,7 +68,7 @@ private static void scheduleUpdate(IEditorPart editorPart, ISonarLintFile sonarL
 
   private static void scheduleUpdate(FileWithDocument fileWithDoc) {
     ISonarLintFile file = fileWithDoc.getFile();
-    if (!file.getProject().isAutoEnabled()) {
+    if (!file.getProject().isAutoEnabled() || !FileExclusionsUtils.shouldAnalyze(file.getProject(), file.getResource())) {
```

Comment:
I'm not comfortable with the use of file.getProject() and file.getResource(). Especially because you are later using their path to decide on the exclusion. Won't it be possible to use the ISonarLintFile::getProjectRelativePath() since this is the "logical" path.